connector-runtime/connector-runtime-spring/src/main/java/io/camunda/connector/runtime/inbound/executable/InboundExecutableRegistryImpl.java

```diff
@@ -0,0 +1,320 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.connector.runtime.inbound.executable;
+
+import io.camunda.connector.api.inbound.Health;
+import io.camunda.connector.api.inbound.ProcessElement;
+import io.camunda.connector.runtime.core.config.InboundConnectorConfiguration;
+import io.camunda.connector.runtime.core.inbound.InboundConnectorDetails;
+import io.camunda.connector.runtime.core.inbound.InboundConnectorElement;
+import io.camunda.connector.runtime.core.inbound.InboundConnectorFactory;
+import io.camunda.connector.runtime.inbound.executable.InboundExecutableEvent.Deactivated;
+import io.camunda.connector.runtime.inbound.executable.RegisteredExecutable.Activated;
+import io.camunda.connector.runtime.inbound.executable.RegisteredExecutable.ConnectorNotRegistered;
+import io.camunda.connector.runtime.inbound.executable.RegisteredExecutable.FailedToActivate;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.function.BiConsumer;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.scheduling.annotation.Scheduled;
+
+public class InboundExecutableRegistryImpl implements InboundExecutableRegistry {
+
+  private final BlockingQueue<InboundExecutableEvent> eventQueue;
+  private final ExecutorService executorService;
+
+  private final BatchExecutableProcessor batchExecutableProcessor;
+
+  private final Map<ProcessElement, UUID> executablesByElement = new ConcurrentHashMap<>();
+  private final Map<UUID, RegisteredExecutable> executables = new HashMap<>();
+
+  private static final Logger LOG = LoggerFactory.getLogger(InboundExecutableRegistryImpl.class);
+
+  private final Map<String, List<String>> deduplicationScopesByType;
+
+  public InboundExecutableRegistryImpl(
+      InboundConnectorFactory connectorFactory, BatchExecutableProcessor batchExecutableProcessor) {
+    this.batchExecutableProcessor = batchExecutableProcessor;
+    this.executorService = Executors.newSingleThreadExecutor();
+    eventQueue = new LinkedBlockingQueue<>();
+    deduplicationScopesByType =
+        connectorFactory.getConfigurations().stream()
+            .collect(
+                Collectors.toMap(
+                    InboundConnectorConfiguration::type,
+                    InboundConnectorConfiguration::deduplicationProperties));
```

Comment:
My assumption is that it would duplicate/reuse the logic used by the factory because we need to be able to fetch it from both env vars and annotation config. So I'm not sure if the benefit of a proper type wins against the consolidated configuration processing :)

Let me follow up on this one in the upcoming PRs with further adjustments